core/src/main/java/org/elasticsearch/search/fetch/subphase/ParentFieldSubFetchPhase.java

```diff
@@ -46,6 +46,11 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
             return;
         }
 
+        if (hitContext.hit().getNestedIdentity() != null) {
+            // skip nested documents as they nested have a join field
```

Comment:
If I'm not mistaken, this would not work if we were able to put a parent/child inner hit under a nested inner hit. Maybe it does not matter since it is not possible today, but I don't like that this fix does not generalize well. Maybe we should instead fix `getParentId` to return `null` when a document does not have a `_parent` field using `LeafReader.getDocsWithField`?